redis/src/main/scala/akka/streams/alpakka/redis/scaladsl/RedisSource.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.streams.alpakka.redis.scaladsl
+
+import java.util.function.Predicate
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import akka.streams.alpakka.redis.{RedisHSet, RedisKeyValue, RedisPubSub}
+import io.lettuce.core.KeyValue
+import io.lettuce.core.api.StatefulRedisConnection
+import io.lettuce.core.pubsub.StatefulRedisPubSubConnection
+import io.lettuce.core.pubsub.api.reactive.ChannelMessage
+import reactor.core.publisher.Flux
+
+import scala.collection.immutable.Seq
+
+object RedisSource {
+
+  def subscribe[K, V](topics: Seq[K],
+                      connection: StatefulRedisPubSubConnection[K, V]): Source[RedisPubSub[K, V], NotUsed] = {
+    connection.reactive().subscribe(topics: _*).subscribe()
```

Comment:
Kafka source materializes to `Control` and the JMS source materializes to `KillSwitch` to allow proper stop of subscription and proper stream shutdown. Shall we do something along those lines?

Reply:
I implemented unsubscribe as part of the flow, so user can unsubscribe when ever he wants/